.github/workflows/test.yml

```diff
@@ -5,6 +5,15 @@ jobs:
   test-core:
     name: Run Core Tests
     runs-on: ubuntu-latest
+    strategy:
+      fail-fast: false
+      matrix:
+        python-version: [3.6, 3.7, 3.8, 3.9]
+    # We want to run on external PRs, but not on our own internal PRs as they'll be run
+    # by the push to the branch. This prevents duplicated runs on internal PRs.
+    # Some discussion of this here: 
+    # https://github.community/t/duplicate-checks-on-push-and-pull-request-simultaneous-event/18012
+    if: github.event_name == 'push' || github.event.pull_request.head.repo.full_name != github.repository
```

Comment:
I think this is a reasonable solution to the problem of actions running twice that [@leonardr mentioned in his comment here](https://github.com/NYPL-Simplified/server_core/pull/1245/files/be15b56c90463e17d6881b5f0272d4eda55fef73#diff-faff1af3d8ff408964a57b2e475f69a6b7c7b71c9978cccc8f471798caac2c88).